diff --git a/node_modules/react-native-progress-circle/src/index.js b/node_modules/react-native-progress-circle/src/index.js
index 50cb271..4df3987 100644
--- a/node_modules/react-native-progress-circle/src/index.js
+++ b/node_modules/react-native-progress-circle/src/index.js
@@ -1,6 +1,7 @@
 import React, { Component } from 'react'
 import PropTypes from 'prop-types'
-import { StyleSheet, View, ViewPropTypes,I18nManager } from 'react-native'
+import { StyleSheet, View,I18nManager } from 'react-native'
+import { ViewPropTypes } from "deprecated-react-native-prop-types";
 
 // compatability for react-native versions < 0.44
 const ViewPropTypesStyle = ViewPropTypes
