diff --git a/node_modules/react-native-vision-camera/android/src/main/java/com/mrousavy/camera/core/VideoPipeline.kt b/node_modules/react-native-vision-camera/android/src/main/java/com/mrousavy/camera/core/VideoPipeline.kt
index ddbec56..9f1537b 100644
--- a/node_modules/react-native-vision-camera/android/src/main/java/com/mrousavy/camera/core/VideoPipeline.kt
+++ b/node_modules/react-native-vision-camera/android/src/main/java/com/mrousavy/camera/core/VideoPipeline.kt
@@ -104,8 +104,12 @@ class VideoPipeline(
         frame.incrementRefCount()
         frameProcessor?.call(frame)
 
-        imageWriter!!.queueInputImage(image)
-
+        try {
+          imageWriter!!.queueInputImage(image)
+        }
+        catch (exception: RuntimeException){
+          // Handling exception
+        }
         frame.decrementRefCount()
       }, CameraQueues.videoQueue.handler)
 
