diff --git a/node_modules/react-native-check-app-install/android/build.gradle b/node_modules/react-native-check-app-install/android/build.gradle
index d374b27..ff72933 100644
--- a/node_modules/react-native-check-app-install/android/build.gradle
+++ b/node_modules/react-native-check-app-install/android/build.gradle
@@ -5,7 +5,6 @@ def safeExtGet(prop, fallback) {
 buildscript {
     repositories {
         google()
-        jcenter()
     }
 
     dependencies {
@@ -35,7 +34,6 @@ android {
 repositories {
     google()
     mavenLocal()
-    jcenter()
     maven {
         // All of React Native (JS, Obj-C sources, Android binaries) is installed from npm
         url "$projectDir/../../../node_modules/react-native/android"
@@ -43,5 +41,5 @@ repositories {
 }
 
 dependencies {
-    compile 'com.facebook.react:react-native:+'
+    implementation 'com.facebook.react:react-native:+'
 }
